drivers/clock_control/clock_control_bl60x.c

```diff
@@ -0,0 +1,942 @@
+	tmp = (tmp & PDS_CLKPLL_PU_FBDV_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_PU_FBDV_POS);
+	tmp = (tmp & PDS_CLKPLL_PU_POSTDIV_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_PU_POSTDIV_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+
+	clock_control_bl60x_clock_settle();
+
+	/* reset couple things one by one? */
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_SDM_RESET_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_SDM_RESET_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_RESET_FBDV_UMSK) | ((uint32_t)(1) << PDS_CLKPLL_RESET_FBDV_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_RESET_FBDV_UMSK) | ((uint32_t)(0) << PDS_CLKPLL_RESET_FBDV_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+
+	tmp = sys_read32(PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+	tmp = (tmp & PDS_CLKPLL_SDM_RESET_UMSK) | ((uint32_t)(0) << PDS_CLKPLL_SDM_RESET_POS);
+	sys_write32(tmp, PDS_BASE + PDS_PU_RST_CLKPLL_OFFSET);
+
+	clock_control_bl60x_set_root_clock(old_rootclk);
+	clock_control_bl60x_clock_settle();
+}
+
+/*
+ * 0: 48M
+ * 1: 120M
+ * 2: 160M
+ * 3: 192M
+ */
+static void clock_control_bl60x_select_PLL(uint8_t pll)
+{
+	uint32_t tmp;
+
+	tmp = sys_read32(GLB_BASE + GLB_CLK_CFG0_OFFSET);
+	tmp = (tmp & GLB_REG_PLL_SEL_UMSK) | (pll << GLB_REG_PLL_SEL_POS);
+	sys_write32(tmp, GLB_BASE + GLB_CLK_CFG0_OFFSET);
+}
+
+static int clock_control_bl60x_clock_trim_32M(void)
+{
+	uint32_t tmp;
+	int err;
+	uint32_t trim, trim_parity;
+	const struct device *efuse = DEVICE_DT_GET_ONE(bflb_efuse);
+
+	err = syscon_read_reg(efuse, EFUSE_RC32M_TRIM_OFFSET, &trim);
+	if (err < 0) {
+		LOG_ERR("Error: Couldn't read efuses: err: %d.\n", err);
+		return err;
+	}
+	if (!((trim >> EFUSE_RC32M_TRIM_EN_POS) & 1)) {
+		LOG_ERR("RC32M trim disabled!");
+		return -EINVAL;
+	}
+
+	trim_parity = (trim >> EFUSE_RC32M_TRIM_PARITY_POS) & 1;
+	trim = (trim & EFUSE_RC32M_TRIM_MSK) >> EFUSE_RC32M_TRIM_POS;
+
+	if (trim_parity != (POPCOUNT(trim) & 1)) {
+		LOG_ERR("Bad trim parity");
+		return -EINVAL;
+	}
+
+	tmp = sys_read32(PDS_BASE + PDS_RC32M_CTRL0_OFFSET);
+	tmp = (tmp & PDS_RC32M_EXT_CODE_EN_UMSK) | 1 << PDS_RC32M_EXT_CODE_EN_POS;
+	tmp = (tmp & PDS_RC32M_CODE_FR_EXT_UMSK) | trim << PDS_RC32M_CODE_FR_EXT_POS;
+	sys_write32(tmp, PDS_BASE + PDS_RC32M_CTRL0_OFFSET);
+
+	clock_control_bl60x_clock_settle();
+
+	return 0;
+}
+
+/* source for most clocks, either XTAL or RC32M */
+static uint32_t clock_control_bl60x_get_xclk(const struct device *dev)
+{
+	uint32_t tmp;
+	const struct clock_control_bl60x_config *config = dev->config;
+
+	tmp = sys_read32(HBN_BASE + HBN_GLB_OFFSET);
+	tmp &= HBN_ROOT_CLK_SEL_MSK;
+	tmp = tmp >> HBN_ROOT_CLK_SEL_POS;
+	tmp &= 1;
+	if (tmp == 0) {
+		return RC32M_FREQ;
```

Comment:
A separate definition seems unnecessary 
```suggestion
		return MHZ(32);
```

Reply:
added a generic value that is shared between all users of rc32m.